### DISCUSSION TITLE
Gracefully shutdown all servers on OS shutdown

### DISCUSSION THREAD

**Discussion by JasonHorkles** [pelican-dev/panel#195] [Ideas]
When shutting down the OS, it would be cool if Pelican gracefully ran the shutdown action on all running servers before allowing the OS to continue

**Comment by Nazgile94** [comment]
but set timelimit to like 60 seconds, so if some container hangs it forces shutdown after XX seconds / minutes  or let admin choose how long to wait for gracefully shutdown

**Comment by lancepioch** [comment]
The servers are running as docker containers. When you turn off your node, it sends the appropriate signals to each container. So this is already handled!
